R/return_linter.R

```diff
@@ -73,13 +74,17 @@ return_linter <- function(
     return_style = c("implicit", "explicit"),
     allow_implicit_else = TRUE,
     return_functions = NULL,
-    except = NULL) {
+    except = NULL,
+    except_regex = NULL) {
   return_style <- match.arg(return_style)
 
-  if (!allow_implicit_else || return_style == "explicit") {
-    except_xpath <- glue("parent::expr[not(
+  check_except <- !allow_implicit_else || return_style == "explicit"
+
+  if (check_except) {
+    except_xpath_fmt <- "parent::expr[not(
       preceding-sibling::expr/SYMBOL[{ xp_text_in_table(union(special_funs, except)) }]
-    )]")
+    )]"
+    if (is.null(except_regex)) except_xpath <- glue(except_xpath_fmt)
```

Comment:
Is simultaneous use of `except` and `except_regex` precluded? Otherwise, I don't understand this condition. If it is, that should be documented.

Reply:
No, using both is fine. If there's `except_regex`, defer compiling this string until run-time when we know the list of possible assignments.

Anyway your other comment is probably the better way to handle this, so resolving.

Reply:
I tried to clarify things with `defer_except` and a comment, hopefully that makes it easier to follow.

Reply:
Thanks, the new flow reads nicely. The `glue(..., bla = bla)` also really helps.